### ISSUE TITLE
'virtual/node' breaks for do_package_deb because of '/'

### ISSUE THREAD

**Issue by fladi** [imyller/meta-nodejs#3]
The char `/` is not allowed in the `Provides:` field in `debian/control`. Perhaps another `PROVIDES` would be useful to make it work with `PACKAGE_CLASSES ?= "package_deb"`.

```
| DEBUG: Executing python function do_package_deb
| dpkg-deb: building package `nodejs-dbg' in `.../tmp/work/cortexa9hf-vfp-neon-poky-linux-gnueabi/nodejs/0.12.0-r1.0/deploy-debs/cortexa9hf-vfp-neon/nodejs-dbg_0.12.0-r1.0_armel.deb'.
| NOTE: Not creating empty archive for nodejs-staticdev-0.12.0-r1.0
| dpkg-deb: building package `nodejs-dev' in `.../tmp/work/cortexa9hf-vfp-neon-poky-linux-gnueabi/nodejs/0.12.0-r1.0/deploy-debs/cortexa9hf-vfp-neon/nodejs-dev_0.12.0-r1.0_armel.deb'.
| dpkg-deb: building package `nodejs-doc' in `.../tmp/work/cortexa9hf-vfp-neon-poky-linux-gnueabi/nodejs/0.12.0-r1.0/deploy-debs/cortexa9hf-vfp-neon/nodejs-doc_0.12.0-r1.0_armel.deb'.
| NOTE: Not creating empty archive for nodejs-locale-0.12.0-r1.0
| dpkg-deb: error: parsing file '.../tmp/work/cortexa9hf-vfp-neon-poky-linux-gnueabi/nodejs/0.12.0-r1.0/packages-split/nodejs/DEBIAN/control' near line 14 package 'nodejs':
|  `Provides' field, invalid package name `virtual/node': character `/' not allowed (only letters, digits and characters `-+._')
| DEBUG: Python function do_package_deb finished
| DEBUG: Python function do_package_write_deb finished
| ERROR: Function failed: dpkg-deb execution failed
```

**Comment by imyller** [comment]
There seems to be a lot of packages providing "virtual/...." so I assumed this is a safe naming convention. 

These are defined so that meta-nodejs and meta-iojs layers can be exchanged without updating depending recipes.

Would it be acceptable to change PROVIDES and RPROVIDES to "node" instead of "virtual/node"?

**Comment by fladi** [comment]
Strange, during my build only nodejs turned up with this error. Using "node" as `(R)PROVIDES` is what I did now for my build. Of all the packages that were built until the point where it failed, only nodejs used the "virtual/" prefix. I'll see if any other package turns up with the same error in my current build.

**Comment by imyller** [comment]
Difference might be that I'm using Ångstrom based distribution without .deb packaging and you are using Poky. Ångstrom layers contain lot of virtual prefixed (R)PROVIDES.

I'm going to commit a change to `PROVIDES = "node"`. I've already checked that it should not collide with commonly used OE packages.

If someone has added `PREFERRED_PROVIDER_virtual/node = "nodejs"` to their `local.conf` they might want to replace it with `PREFERRED_PROVIDER_node`

**Comment by fladi** [comment]
You could have left `PROVIDES` with "virtual/node". It's only `RPROVIDES` that is used in the package generation.

**Comment by imyller** [comment]
We're both too fast. Did not see your pull request coming ;) 

Will fix.

**Comment by fladi** [comment]
I should have sent my comment first and do the pull-request later :-)

Kudos for fixing it tough! Keep up the great work!

**Comment by imyller** [comment]
Thank you. I'll fix the meta-iojs too (imyller/meta-iojs#1)
